doc/nom_recipes.md

```diff
@@ -0,0 +1,235 @@
+
+## Comments
+
+### `// C++/EOL-style comments`
+
+This version does not consume the newline character and returns an output of `()`.
+
+```rust
+pub fn peol_comment<'a, E: ParseError<&'a str>>(i: &'a str) -> IResult<&'a str, (), E>
+{
+  map(pair(char('%'), is_not("\n\r")),
+      |_| () // Output is thrown away.
+  )(i)
+}
+```
+
+### `/* C-style comments */`
+
+Inline comments surrounded with sentinel tags `(*` and `*)`. This version returns an output of `()` and does not handle nested comments.
+
+```rust
+pub fn pinline_comment<'a, E: ParseError<&'a str>>(i: &'a str) -> IResult<&'a str, (), E> {
+  map(
+    tuple((
+      tag("(*"),
+      take_until("*)"),
+      tag("*)")
+    )),
+    |_| () // Output is thrown away.
+  )(i)
+}
+```
+
+## Identifiers
+
+### `Rust-Style Identifiers`
+
+Parsing identifiers that may start with a letter (or underscore) and may contain underscores, letters and numbers may be parsed like this:
+
+```rust
+pub fn identifier(input: &str) -> IResult<&str, &str> {
+  recognize(
+    pair(
+      alt((alpha1, tag("_"))),
+      many0(alt((alphanumeric1, tag("_"))))
+    )
+  )(input)
+}
+```
+
+Let's say we apply this to the identifier `hello_world123abc`.
+The first `alt` parser would recognize `hello_world`. The `pair` combinator ensures that `123abc` will be piped to the next `alphanumeric0` parser, which recognizes every remaining character.
+However, the `pair` combinator returns a tuple of the results of its sub-parsers. The `recognize` parser produces a `&str` of the input text that was parsed.
+
+## Literal Values
+
+### Escaped Strings
+
+This is [one of the examples](https://github.com/Geal/nom/blob/master/examples/string.rs) in the examples directory.
+
+### Integers
+
+The following recipes all return string slices rather than integer values. How to obtain an integer value instead is demonstrated for hexadecimal integers. The others are similar.
+
+The parsers allow the grouping character `_`, which allows one to group the digits by byte, for example: `0xA4_3F_11_28`. If you prefer to exclude the `_` character, the lambda to convert from a string slice to an integer value is slightly simpler. You can also strip the `_` from the string slice that is returned, which is demonstrated in the second hexdecimal number parser.
+
+If you wish to limit the number of digits in a valid integer literal, replace `many1` with `many_m_n` in the recipes: `many1(one_of("_01"))` becomes `many_m_n(1, 16, one_of("_01"))` for binary numbers having at most 16 bits.
+
+#### Hexadecimal
+
+The parser outputs the string slice of the digits without the leading `0x`/`0X`.
+
+```rust
+fn hexadecimal<'a, E: ParseError<&'a str>>(input: &'a str) -> IResult<&'a str,&'a str, E> {
```

Comment:
Nit, but a bunch of these function signatures are missing a space after `IResult<&'a str,`